sycl/source/detail/kernel_impl.hpp

```diff
@@ -257,6 +261,20 @@ kernel_impl::get_info(const device &Device,
       getPlugin());
 }
 
+template <>
+inline typename ext::oneapi::experimental::info::kernel_queue_specific::
+    max_num_work_group_sync::return_type
+    kernel_impl::ext_oneapi_get_info<
+        ext::oneapi::experimental::info::kernel_queue_specific::
+            max_num_work_group_sync>(const queue &Queue) const {
+  const auto &Plugin = getPlugin();
+  const auto &Handle = getHandleRef();
+  pi_uint32 GroupCount = 0;
+  Plugin->call<PiApiKind::piextKernelSuggestMaxCooperativeGroupCount>(
+      Handle, &GroupCount);
+  return GroupCount;
+}
+
```

Comment:
I'm not sure if this specialization should live here and not in the extension header.